wal/wal.go

```diff
@@ -41,6 +41,10 @@ const (
 
 	// the owner can make/remove files inside the directory
 	privateDirMode = 0700
+
+	// the expected size of each wal segment file.
+	// the actual size might be bigger than it.
+	segmentSizeBytes = 8 * 1000 * 1000 // 8MB
```

Comment:
The size of most entries should be smaller than 1kb. 8MB can store around 10K entries. We will per-allocate the disk space to gain some performance and also to fail beforehand if you do not have enough disk space. 8MB is large enough to gain the performance.